code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -156,6 +158,13 @@ protected void setDns(Instance instance, Map<String, Object> labels, Map<Object,
             }
         }
     }
+    
+    protected void setSystemLabel(Instance instance, Map<String, Object> labels, Map<Object, Object> data) {
+        if(instance.getSystem()) {
```

Comment:
Since getSystem() returns a Boolean oject and not the primitive boolean, this can cause a null pointer exception.

Protect against it by making the check `if (Boolean.TRUE.equals(instance.getSystem())) { ... }`